File: Python/OOP/Inheritance/Hybrid inheritance (example 1).py
class Institute:
    institute_name = "Institute of Engineering"  # just a random name


class Tacher:
    teacher_name = "Guido van Rossum"
    subject = "Python Programming"


class Student:
    student_name = "Abdullah Al Naiem"


# inherit multiple classes at once and combind into a method
class Multiple_Inh(Institute, Tacher, Student):
    def displayInfo(self):
        print(self.institute_name)
        print("Teacher: ", self.teacher_name)

# Hybrid inheritance is a combination of multiple inheritance and multilevel inheritance. 
# A class is derived from two classes as in multiple inheritance.
# institute, teacher and student class inherited by "multiple" and it's inherited by "hybrid" class
class Hybrid_Inh(Multiple_Inh):
    def displayInfo(self):
        print("Subject: ", self.subject)
        print("Student: ", self.student_name)
        

teacher = Multiple_Inh()
teacher.displayInfo()

student = Hybrid_Inh()
student.displayInfo()

File: C++/OOP/Inheritance/Hybrid inheritance (example 1).cpp
#include <iostream>
using namespace std;

class Institute {
    public:
        string institute_name = "Institute of Engineering" ; // just a random name
};
class Teacher {
    public:
        string teacher_name = "Bjarne Stroustrup";
        string subject = "C Plus Plus";
};
class Student {
    public: 
        string student_name = "Abdullah Al Naiem";
};

class Multiple_Inh : public Institute, public Teacher, public Student {
    public:
        void displayInfo(){
            cout << institute_name << endl;
            cout << "Teacher: " << teacher_name << endl;
        }
};

class Hybrid_Inh : public Multiple_Inh {
    public:
        void displayInfo(){
            cout << "Subject: " << subject << endl;
            cout << "Student: " << student_name << endl;
        }
};

int main(){
    Multiple_Inh teacher;
    teacher.displayInfo();

    Hybrid_Inh student;
    student.displayInfo();
}